scanamo/src/main/scala/org/scanamo/ops/ScanamoOpsA.scala

```diff
@@ -1,48 +1,80 @@
+final case class Update[A](req: ScanamoUpdateRequest, res: ScanamoResult[UpdateItemResult] => A) extends ScanamoOpsA[A]
+final case class Fail(t: Throwable) extends ScanamoOpsA[Nothing]
 
 object ScanamoOps {
 
   import cats.free.Free.liftF
 
-  def put(req: ScanamoPutRequest): ScanamoOps[PutItemResult] = liftF[ScanamoOpsA, PutItemResult](Put(req))
-  def conditionalPut(req: ScanamoPutRequest): ScanamoOps[Either[ConditionalCheckFailedException, PutItemResult]] =
-    liftF[ScanamoOpsA, Either[ConditionalCheckFailedException, PutItemResult]](ConditionalPut(req))
-  def get(req: GetItemRequest): ScanamoOps[GetItemResult] = liftF[ScanamoOpsA, GetItemResult](Get(req))
-  def delete(req: ScanamoDeleteRequest): ScanamoOps[DeleteItemResult] =
-    liftF[ScanamoOpsA, DeleteItemResult](Delete(req))
-  def conditionalDelete(
+  def put(req: ScanamoPutRequest): ScanamoOps[ScanamoResult[PutItemResult]] =
+    liftF[ScanamoOpsA, ScanamoResult[PutItemResult]](Put(req, identity))
+  def get(req: GetItemRequest): ScanamoOps[ScanamoResult[GetItemResult]] =
+    liftF[ScanamoOpsA, ScanamoResult[GetItemResult]](Get(req, identity))
+  def delete(
     req: ScanamoDeleteRequest
-  ): ScanamoOps[Either[ConditionalCheckFailedException, DeleteItemResult]] =
-    liftF[ScanamoOpsA, Either[ConditionalCheckFailedException, DeleteItemResult]](ConditionalDelete(req))
-  def scan(req: ScanamoScanRequest): ScanamoOps[ScanResult] = liftF[ScanamoOpsA, ScanResult](Scan(req))
-  def query(req: ScanamoQueryRequest): ScanamoOps[QueryResult] = liftF[ScanamoOpsA, QueryResult](Query(req))
+  ): ScanamoOps[ScanamoResult[DeleteItemResult]] =
+    liftF[ScanamoOpsA, ScanamoResult[DeleteItemResult]](Delete(req, identity))
+  def scan(req: ScanamoScanRequest): ScanamoOps[ScanResult] = liftF[ScanamoOpsA, ScanResult](Scan(req, identity))
+  def query(req: ScanamoQueryRequest): ScanamoOps[QueryResult] = liftF[ScanamoOpsA, QueryResult](Query(req, identity))
   def batchWrite(req: BatchWriteItemRequest): ScanamoOps[BatchWriteItemResult] =
-    liftF[ScanamoOpsA, BatchWriteItemResult](BatchWrite(req))
+    liftF[ScanamoOpsA, BatchWriteItemResult](BatchWrite(req, identity))
   def batchGet(req: BatchGetItemRequest): ScanamoOps[BatchGetItemResult] =
-    liftF[ScanamoOpsA, BatchGetItemResult](BatchGet(req))
-  def update(req: ScanamoUpdateRequest): ScanamoOps[UpdateItemResult] =
-    liftF[ScanamoOpsA, UpdateItemResult](Update(req))
-  def conditionalUpdate(
+    liftF[ScanamoOpsA, BatchGetItemResult](BatchGet(req, identity))
+  def update(
     req: ScanamoUpdateRequest
-  ): ScanamoOps[Either[ConditionalCheckFailedException, UpdateItemResult]] =
-    liftF[ScanamoOpsA, Either[ConditionalCheckFailedException, UpdateItemResult]](ConditionalUpdate(req))
+  ): ScanamoOps[ScanamoResult[UpdateItemResult]] =
+    liftF[ScanamoOpsA, ScanamoResult[UpdateItemResult]](Update(req, identity))
+//TODO: Does there need to be a fail lift?
```

Comment:
Yes.

Reply:
This is actually going to be useful in `ConditionExpression`. Let's imagine this function is called `fail`.

Reply:
Scratch that, I was wrong. But yeah, please add the `fail` combinator anyway 🤗

Reply:
Added